Framework/Geometry/inc/MantidGeometry/Instrument/Container.h

```diff
@@ -79,14 +79,15 @@ class MANTID_GEOMETRY_DLL Container final : public IObject {
   int getPointInObject(Kernel::V3D &point) const override {
     return m_shape->getPointInObject(point);
   }
-  Kernel::V3D generatePointInObject(Kernel::PseudoRandomNumberGenerator &rng,
-                                    const size_t i) const override {
-    return m_shape->generatePointInObject(rng, i);
-  }
-  Kernel::V3D generatePointInObject(Kernel::PseudoRandomNumberGenerator &rng,
-                                    const BoundingBox &activeRegion,
-                                    const size_t i) const override {
-    return m_shape->generatePointInObject(rng, activeRegion, i);
+  bool generatePointInObject(Kernel::PseudoRandomNumberGenerator &rng,
```

Comment:
We should use `boost:optional<Kernel:V3D>` here instead of an out parameter

Reply:
OK done. Also done in parent IObject.h + other sibling classes MeshObject.h, MeshObject2D.h